src/main/java/net/minecraftforge/event/entity/player/PlayerEvent.java

```diff
@@ -510,4 +511,65 @@ public PlayerChangedDimensionEvent(PlayerEntity player, RegistryKey<World> fromD
             return this.toDim;
         }
     }
+
+    /**
+     * Fired when the game type of a server player is changed to a unique value before listeners are fired.
```

Comment:
I have no idea what you mean to say with this sentence. "changed to a unique value"? And what listeners are you talking about?
You also need to add a note about what happens when the event is cancelled.